examples/extformer_moe/extformer_moe_enso_train.py

```diff
@@ -0,0 +1,224 @@
+    ]
+
+    # # init optimizer and lr scheduler
+    lr_scheduler_cfg = dict(cfg.TRAIN.lr_scheduler)
+    lr_scheduler = ppsci.optimizer.lr_scheduler.Cosine(
+        **lr_scheduler_cfg,
+        iters_per_epoch=ITERS_PER_EPOCH,
+        eta_min=cfg.TRAIN.min_lr_ratio * cfg.TRAIN.lr_scheduler.learning_rate,
+        warmup_epoch=int(0.2 * cfg.TRAIN.epochs),
+    )()
+    optimizer = paddle.optimizer.AdamW(
+        lr_scheduler, parameters=optimizer_grouped_parameters
+    )
+        
+    # initialize solver, eval_freq: int = 1
+    solver = ppsci.solver.Solver(
+        model,
+        constraint,
+        cfg.output_dir,
+        optimizer,
+        lr_scheduler,
+        cfg.TRAIN.epochs,
+        ITERS_PER_EPOCH,
+        update_freq=cfg.TRAIN.update_freq,
+        eval_during_train=cfg.TRAIN.eval_during_train,
+        seed=cfg.seed,
+        device=f'gpu:{cfg.DEVICE}' if cfg.DEVICE != -1 else 'cpu', 
+        validator=validator,
+        compute_metric_by_batch=cfg.EVAL.compute_metric_by_batch,
+        eval_with_no_grad=cfg.EVAL.eval_with_no_grad,
+    )
+
+    # train model
+    solver.train()
+    # evaluate after finished training
+    solver.eval()
+
+
+def evaluate(cfg: DictConfig):
+    # set eval dataloader config
+    eval_dataloader_cfg = {
+        "dataset": {
+            "name": "ExtMoEENSODataset",
+            "data_dir": cfg.FILE_PATH,
+            "input_keys": cfg.MODEL.input_keys,
+            "label_keys": cfg.DATASET.label_keys,
+            "in_len": cfg.DATASET.in_len,
+            "out_len": cfg.DATASET.out_len,
+            "in_stride": cfg.DATASET.in_stride,
+            "out_stride": cfg.DATASET.out_stride,
+            "train_samples_gap": cfg.DATASET.train_samples_gap,
+            "eval_samples_gap": cfg.DATASET.eval_samples_gap,
+            "normalize_sst": cfg.DATASET.normalize_sst,
+            "training": "test",
+        },
+        "batch_size": cfg.EVAL.batch_size,
+    }
+
+    sup_validator = ppsci.validate.SupervisedValidator(
+        eval_dataloader_cfg,
+        loss=ppsci.loss.FunctionalLoss(enso_metric.train_extformer_moe_func),
+        metric={
+            "rmse": ppsci.metric.FunctionalMetric(enso_metric.eval_rmse_func),
+        },
+        name="Sup_Validator",
+    )
+    validator = {sup_validator.name: sup_validator}
+
+    moe_config = OmegaConf.to_object(cfg.MOE)
+    rnc_config = OmegaConf.to_object(cfg.RNC)
+    model = ppsci.arch.ExtFormerMoECuboid(
+        **cfg.MODEL, moe_config=moe_config, rnc_config=rnc_config
+    )
+
+    # initialize solver
+    solver = ppsci.solver.Solver(
+        model,
+        output_dir=cfg.output_dir,
+        log_freq=cfg.log_freq,
+        seed=cfg.seed,
+        device=f'gpu:{cfg.DEVICE}' if cfg.DEVICE != -1 else 'cpu', 
+        validator=validator,
+        pretrained_model_path=cfg.EVAL.pretrained_model_path,
+        compute_metric_by_batch=cfg.EVAL.compute_metric_by_batch,
+        eval_with_no_grad=cfg.EVAL.eval_with_no_grad,
+    )
```

Comment:
如果device这个参数可以删除的话，Solver的初始化可以简化成如下代码（只传递cfg）
```suggestion
    solver = ppsci.solver.Solver(
        model,
        validator=validator,
        cfg=cfg,
    )
```

Reply:
可能不太方便化简，solver 中利用 cfg 自动填充 attributes，但很多是 yaml 文件中没有定义的 attributes

Reply:
> 可能不太方便化简，solver 中利用 cfg 自动填充 attributes，但很多是 yaml 文件中没有定义的 attributes

Omegaconf是一个可修改的opendict，所以cfg带有yaml文件没有定义的attr是没有问题的。